src/lib/optimizer/optimizer.cpp

```diff
@@ -104,14 +105,16 @@ std::shared_ptr<Optimizer> Optimizer::create_default_optimizer() {
 
   optimizer->add_rule(std::make_unique<BetweenCompositionRule>());
 
-  // Position the predicates after the JoinOrderingRule ran. The JOR manipulates predicate placement as well, but
-  // for now we want the PredicateReorderingRule to have the final say on predicate positions
-  optimizer->add_rule(std::make_unique<PredicatePlacementRule>());
-
   optimizer->add_rule(std::make_unique<PredicateSplitUpRule>());
 
   optimizer->add_rule(std::make_unique<SubqueryToJoinRule>());
 
+  optimizer->add_rule(std::make_unique<SemiJoinReductionRule>());
+
+  // Position the predicates after the JoinOrderingRule ran. The JOR manipulates predicate placement as well, but
+  // for now we want the PredicateReorderingRule to have the final say on predicate positions
+  optimizer->add_rule(std::make_unique<PredicatePlacementRule>());
```

Comment:
That comment was just moved, wasn't it? This shouldn't have any impact on the review of this PR, but the general idea is that because the JoinOrderingRule rebuilds the plan, it has to install some order. If we ran the PredicatePlacementRule first, its order would get ignored. As the PredicatePlacementRule is more powerful, it will probably stay like this forever. I'd remove the "for now".